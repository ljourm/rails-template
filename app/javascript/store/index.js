import Vue from "vue"
import Vuex from "vuex"
Vue.use(Vuex)

import base from "./modules/base"
import api from "./modules/api"
import user from "./modules/user"
import users from "./modules/users"
import roles from "./modules/roles"

export default new Vuex.Store({
  strict: process.env.NODE_ENV === "development",
  modules: {
    base: base,
    api: api,
    user: user,
    users: users,
    roles: roles,
  },
})
