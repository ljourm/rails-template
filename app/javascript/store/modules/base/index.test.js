import store from "./index"

describe("store", () => {
  test("snapshot", () => {
    expect(store).toMatchSnapshot()
  })
})
