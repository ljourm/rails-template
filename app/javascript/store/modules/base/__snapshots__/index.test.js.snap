// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`store snapshot 1`] = `
Object {
  "actions": Object {
    "logout": [Function],
    "redirectToLogin": [Function],
    "setErrorMessage": [Function],
  },
  "mutations": Object {
    "CLEAR_ERROR_MESSAGE": [Function],
    "SET_ERROR_MESSAGE": [Function],
    "SET_LOADING": [Function],
  },
  "namespaced": true,
  "state": Object {
    "errorMessage": null,
    "loading": false,
  },
}
`;
